src/filters/ws.rs

```diff
@@ -134,10 +140,11 @@ where
     fn into_response(self) -> Response {
         let on_upgrade = self.on_upgrade;
         let config = self.ws.config;
-        let fut = self
-            .ws
-            .body
-            .on_upgrade()
+        let mut resp = Response::new(self.ws.body);
+        if let Some(on_upgrade) = self.ws.on_upgrade {
+            resp.extensions_mut().insert(on_upgrade);
```

Comment:
If you already have the extension, you don't need to insert it here to remove it again. (`on_upgrade` is the same as `fut`).

Reply:
@seanmonstar I think I see what you mean, and try to simplify it to the best of my ability. Please have a look.